pkg/scheduler/scheduler.go

```diff
@@ -338,7 +346,7 @@ func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, e ent
 	log.V(2).Info("Workload re-queued", "workload", klog.KObj(e.Obj), "clusterQueue", klog.KRef("", e.ClusterQueue), "queue", klog.KRef(e.Obj.Namespace, e.Obj.Spec.QueueName), "requeueReason", e.requeueReason, "added", added)
 
 	if e.status == notNominated {
-		err := workload.UpdateStatus(ctx, s.client, e.Obj, kueue.WorkloadAdmitted, metav1.ConditionFalse, "Pending", e.inadmissibleMsg, constants.AdmissionName)
+		err := workload.UnsetAdmissionWithCondition(ctx, s.client, e.Obj, metav1.ConditionFalse, "Pending", e.inadmissibleMsg)
```

Comment:
This particular call should either use SSA, but include a ResourceVersion, or just use Update. Either method would guarantee that we don't accidentally evict a workload that was recently admitted.

Reply:
I tried with ResourceVersion. It fails anyway. I used checking if somehow of the workload in already assumed in the cache and drop it in "nominate" function.